integration-test-groups/aws2/aws2-sqs-sns/src/test/java/org/apache/camel/quarkus/component/aws2/sqs/it/Aws2SqsSnsTest.java

```diff
@@ -40,28 +45,157 @@
+
+        final String msg = sendSingleMessageToQueue(queueName);
+        awaitMessageFromQueue(msg, queueName);
+    }
+
+    private String[] listQueues() {
+        return RestAssured.get("/aws2-sqs-sns/sqs/queues")
                 .then()
                 .statusCode(200)
                 .extract()
                 .body().as(String[].class);
-        Assertions.assertTrue(Stream.of(queues).anyMatch(url -> url.contains(queueName)));
+    }
+
+    @Test
+    public void sqsDeleteMessage() {
+        final String qName = getPredefinedQueueName();
+        final String msg = sendSingleMessageToQueue(qName);
+        final String receipt = receiveReceiptOfMessageFromQueue(qName);
+        deleteMessageFromQueue(qName, receipt);
+        Assertions.assertNotEquals(receiveMessageFromQueue(qName), msg);
+    }
+
+    private String getPredefinedQueueName() {
+        return ConfigProvider.getConfig().getValue("aws-sqs.queue-name", String.class);
+    }
 
+    private String sendSingleMessageToQueue(String queueName) {
         final String msg = "sqs" + UUID.randomUUID().toString().replace("-", "");
         RestAssured.given()
                 .contentType(ContentType.TEXT)
                 .body(msg)
-                .post("/aws2-sqs-sns/sqs/send")
+                .post("/aws2-sqs-sns/sqs/send/" + queueName)
                 .then()
                 .statusCode(201);
+        return msg;
+    }
+
+    private String receiveReceiptOfMessageFromQueue(String queueName) {
+        return RestAssured.get("/aws2-sqs-sns/sqs/receive/receipt/" + queueName)
+                .then()
+                .statusCode(200)
+                .extract()
+                .body()
+                .asString();
+    }
+
+    private void deleteMessageFromQueue(String queueName, String receipt) {
+        RestAssured.delete("/aws2-sqs-sns/sqs/delete/message/" + queueName + "/" + receipt)
+                .then()
+                .statusCode(200);
+    }
+
+    @Test
+    public void sqsAutoCreateDelayedQueue() {
+        final String qName = "delayQueue";
+        final int delay = 10;
+        createDelayQueueAndVerifyExistence("delayQueue", 10);
+        final String msgSent = sendSingleMessageToQueue(qName);
+        Instant start = Instant.now();
+        awaitMessageFromQueue(msgSent, qName);
+        Assertions.assertTrue(Duration.between(start, Instant.now()).getSeconds() >= delay);
+        deleteQueue(qName);
+    }
 
+    private void createDelayQueueAndVerifyExistence(String queueName, int delay) {
+        final String[] queues = RestAssured.get("/aws2-sqs-sns/sqs/queue/autocreate/delayed/" + queueName + "/" + delay)
+                .then()
+                .statusCode(200)
+                .extract()
+                .body()
+                .as(String[].class);
+        Assertions.assertTrue(Stream.of(queues).anyMatch(url -> url.contains(queueName)));
+    }
+
+    private void awaitMessageFromQueue(String expectedContent, String queueName) {
```

Comment:
Maybe `awaitMessageFromQueueWithExpectedContent(...)` would be clearer ?

Reply:
yes, thank you. I've changed it to `awaitMessageWithExpectedContentFromQueue(...)` so that it explains both parameters